enhancements/bundle-validation.md

```diff
@@ -0,0 +1,222 @@
+
+The new validation library can address each of the aforementioned shortcomings by the following:
+
+- **Issue 1:** The library will be written in Golang and maintained as part of [operator-framework/api](https://github.com/operator-framework/api). The bundle structs will be moved to _operator-framework/api_ and act as the source of truth for the structure of these objects by the [OLM](https://github.com/operator-framework/operator-lifecycle-manager), [Operator-SDK](https://github.com/operator-framework/operator-sdk), and this validation library. This change will synchronize the state of the bundle structure across all projects making it easier to maintain, import, and serialize.
+
+- **Issue 2:** Rather than rewriting the courier project to provide static analysis for bundles provided as Catalog Images, the new library can be written and then imported by tools like _scorecard_ to perform validation.
+
+### Goals
+
+- Define a set of rules/validations so that bundles with their specific set of configurations can be validated by checking the required information and components.
+
+### Non-Goals
+
+- The validation compares different versions of bundles from the same package.
+ 
+- The validation considers the validity of a bundle in the context of an index, which should be part of [operator-registry](https://github.com/operator-framework/operator-registry).
+
+- The validation guarantees the successful installation of a bundle (an operator).
+
+## Proposal
+
+### User Stories
+
+As an Operator developer or maintainer, I would like to discover my missing vital metadata and ensure that my Operator bundle is well-formed.
+
+As a cluster administrator or release pipeline owner, I would want to be aware of malformed Operator bundles so that I can prevent them from being released into a catalog for cluster tenants to install.
+
+As a community contributor to the bundle validation library, I would want to understand the workflow of contributing to the core validation strategies or customized validations as part of an extended package.
+
+### Implementation Details/Notes/Constraints
+
+- The bundle validation library should list out errors based on static scanning of a bundle directory and locate their root causes if applicable. 
+
+- Much like project [flak8](https://pypi.org/project/flake8/) where a [list of validations](https://www.flake8rules.com/) can be individually enabled/disabled, individual bundle validating rule should also be separable. Ultimately, tools that are interested in importing the validation library should be able to cherry-pick validations used for different scenarios and configurations. 
+
+- The library should provide all validations currently available within the operator-courier.
+
+- The bundle validation library will be used to validate OLM/registry manifests and expose an interface to add validators for any scheme. 
+
+- Errors and alerting messages should be formatted in a standardized fashion.
```

Comment:
We need to make this output multiple ways. 

Can I suggest a "human readable" output and a JSON output. Whatever is built here needs to be consumed programmatically in the pipeline case.